### PR TITLE
【Hackathon 6th No.24 】 为 paddle.nn.LSTM/RNNBase 功能增强

### PR DESCRIPTION
**Pull request by Asthestarsfalll**
Update docs of LSTM, LSTMCell and RNNBase

- https://github.com/PaddlePaddle/Paddle/pull/63284

### CODE REVIEW

**Comment by zhwesky2010** [comment]
@Asthestarsfalll 附一下Paddle的PR链接吧

File: docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.RNNBase.md [hunk 1]
```diff
@@ -12,52 +12,28 @@ torch.nn.RNNBase(mode: str, input_size: int, hidden_size: int,
...
+        bias_ih_attr=None, bias_hh_attr=None, proj_size=0)
 ```
 
 两者功能一致但参数不一致，部分参数名不同，具体如下：
 ### 参数映射
 
 | PyTorch       | PaddlePaddle | 备注                                                   |
 | ------------- | ------------ | ------------------------------------------------------ |
-| mode          | mode         | 表示 `RNN` 模型的类型,torch 取值为 `'LSTM', 'GRU', 'RNN_TANH', 'RNN_RELU`，paddle 取值为 `'LSTM', 'GRU'`，需要转写。|
+| mode          | mode         | 表示 `RNN` 模型的类型, 取值为 `'LSTM', 'GRU', 'RNN_TANH', 'RNN_RELU`，二者一致，无需转写。|
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.RNNBase.md]: 最后面改成 参数完全一致

File: docs/api/paddle/nn/LSTM_cn.rst [hunk 2]
```diff
@@ -40,12 +44,13 @@ LSTM
...
     - **num_layers** (int，可选) - 循环网络的层数。例如，将层数设为 2，会将两层 GRU 网络堆叠在一起，第二层的输入来自第一层的输出。默认为 1。
     - **direction** (str，可选) - 网络迭代方向，可设置为 forward 或 bidirect（或 bidirectional）。foward 指从序列开始到序列结束的单向 GRU 网络方向，bidirectional 指从序列开始到序列结束，又从序列结束到开始的双向 GRU 网络方向。默认为 forward。
-    - **time_major** (bool，可选) - 指定 input 的第一个维度是否是 time steps。如果 time_major 为 True，则 Tensor 的形状为[time_steps,batch_size,input_size]，否则为[batch_size,time_steps,input_size]。`time_steps` 指输入序列的长度。默认为 False。
+    - **time_major** (bool，可选) - 指定 input 的第一个维度是否是 time steps。如果 time_major 为 True，则 Tensor 的形状为[time_steps, batch_size, input_size]，否则为[batch_size, time_steps, input_size]。`time_steps` 指输入序列的长度。默认为 False。
     - **dropout** (float，可选) - dropout 概率，指的是除第一层外每层输入时的 dropout 概率。范围为[0, 1]。默认为 0。
     - **weight_ih_attr** (ParamAttr，可选) - weight_ih 的参数。默认为 None。
     - **weight_hh_attr** (ParamAttr，可选) - weight_hh 的参数。默认为 None。
     - **bias_ih_attr** (ParamAttr，可选) - bias_ih 的参数。默认为 None。
     - **bias_hh_attr** (ParamAttr，可选) - bias_hh 的参数。默认为 None。
+    - **proj_size** (int，可选) - 若大于 0，则会使用投影层将隐状态隐射到指定大小，其值必须小于 `hidden_size` 。默认为 0。
```
Comment (zhwesky2010) [inline docs/api/paddle/nn/LSTM_cn.rst:53]: torch有这个限制吗？`proj_size < hidden_size`

**Comment by zhwesky2010** [comment]
注：该PR对应的Paddle repo的修改链接为：https://github.com/PaddlePaddle/Paddle/pull/63284

File: docs/api/paddle/nn/LSTM_cn.rst (hunk 2 above)
Comment (Asthestarsfalll) [reply docs/api/paddle/nn/LSTM_cn.rst:53]: 是的

File: docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.RNNBase.md (hunk 1 above)
Comment (Asthestarsfalll) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.RNNBase.md]: 已修改

**Review by zhwesky2010 — APPROVED** [review]

File: docs/api/paddle/nn/LSTMCell_cn.rst [hunk 3]
```diff
@@ -23,6 +23,10 @@ LSTMCell
         h_{t} &= o_{t} * \tanh (c_{t})\\
         y_{t} &= h_{t}
 
+若设置了 `proj_size`，隐状态 :math:`\h_{t}` 将会被映射到指定维度：
+
+.. math::
+        h_{t} & = h_{t} * W_{proj}
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/LSTMCell_cn.rst]:
```suggestion
若设置了 `proj_size`，隐状态 :math:`h_{t}` 将会被映射到指定维度：

.. math::
        h_{t} = h_{t} * W_{proj\_size}
```

File: docs/api/paddle/nn/LSTMCell_cn.rst [hunk 4]
```diff
@@ -41,6 +45,7 @@ LSTMCell
     - **weight_hh_attr** (ParamAttr，可选) - weight_hh 的参数。默认为 None。
     - **bias_ih_attr** (ParamAttr，可选) - bias_ih 的参数。默认为 None。
     - **bias_hh_attr** (ParamAttr，可选) - bias_hh 的参数。默认为 None。
+    - **proj_size** (int，可选) - 若大于 0，则会使用投影层将隐状态隐射到指定大小，其值必须小于 `hidden_size` 。默认为 0。
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/LSTMCell_cn.rst:48]: 对应的 `.. py:class:: paddle.nn.LSTMCell(input_size, hidden_size, weight_ih_attr=None, weight_hh_attr=None, bias_ih_attr=None, bias_hh_attr=None, name=None)` 也加一下 `proj_size` 参数

File: docs/api/paddle/nn/LSTM_cn.rst [hunk 5]
```diff
@@ -29,6 +29,10 @@ LSTM
 
         y_{t} & = h_{t}
 
+若设置了 `proj_size`，隐状态 :math:`\h_{t}` 将会被映射到指定维度：
+
+.. math::
+        h_{t} & = h_{t} * W_{proj}
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/LSTM_cn.rst]:
```suggestion
若设置了 `proj_size`，隐状态 :math:`h_{t}` 将会被映射到指定维度：

.. math::
        h_{t} = h_{t} * W_{proj\_size}
```

File: docs/api/paddle/nn/LSTMCell_cn.rst (hunk 3 above)
Comment (Asthestarsfalll) [reply docs/api/paddle/nn/LSTMCell_cn.rst]: 已更新～

File: docs/api/paddle/nn/LSTMCell_cn.rst [hunk 6]
```diff
@@ -23,6 +23,10 @@ LSTMCell
         h_{t} &= o_{t} * \tanh (c_{t})\\
         y_{t} &= h_{t}
 
+若设置了 `proj_size`，隐状态 :math:`\h_{t}` 将会被映射到指定维度：
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/LSTMCell_cn.rst]:
```suggestion
若设置了 `proj_size`，隐状态 :math:`h_{t}` 将会被映射到指定维度：
```

File: docs/api/paddle/nn/LSTM_cn.rst [hunk 7]
```diff
@@ -29,6 +29,10 @@ LSTM
 
         y_{t} & = h_{t}
 
+若设置了 `proj_size`，隐状态 :math:`\h_{t}` 将会被映射到指定维度：
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/LSTM_cn.rst]:
```suggestion
若设置了 `proj_size`，隐状态 :math:`h_{t}` 将会被映射到指定维度：
```

**Review by sunzhongkai588 — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]
